pkg/hooks/audit_logger_sqlite.go

```diff
@@ -0,0 +1,137 @@
+package hooks
+
+import (
+	"database/sql"
+
+	_ "github.com/mattn/go-sqlite3"
+	"github.com/square/p2/pkg/logging"
+	"github.com/square/p2/pkg/util"
+)
+
+type SQLiteAuditLogger struct {
+	sqlite *sql.DB
+	logger *logging.Logger
+}
+
+func NewSQLiteAuditLogger(sqliteDBPath string, logger *logging.Logger) (*SQLiteAuditLogger, error) {
+	db, err := sql.Open("sqlite3", sqliteDBPath)
+	if err != nil {
+		return nil, err
+	}
+	if db == nil {
+		return nil, util.Errorf("Unable to create a sqlite connection.")
+	}
+	al := &SQLiteAuditLogger{
+		sqlite: db,
+		logger: logger,
+	}
+	if err := al.ensureMigrated(); err != nil {
+		return nil, err
+	}
+	return al, nil
+}
+
+func (al *SQLiteAuditLogger) log(ctx *HookExecContext, success bool) {
+	env := ctx.env
+	stmt := `INSERT INTO hook_results(
+pod_id,
+pod_unique_key,
+hook_name,
+hook_stage,
+success
+) VALUES(?,?,?,?,?)`
+	podID := env.HookedPodIDEnvVar
+	podUniqueKey := env.HookedPodUniqueKeyEnvVar
+	hookName := ctx.Name
+	hookStage := env.HookEventEnvVar
+	// Boolean values are represented by TINYINTs in SQL. Adapt.
+	dbSuccess := 0
+	if !success {
+		dbSuccess = 1
+	}
+
+	_, err := al.sqlite.Exec(stmt, podID, podUniqueKey, hookName, hookStage, dbSuccess)
+	if err != nil {
+		al.logger.Errorln("nope: %v", err)
+	}
+}
+
+func (al *SQLiteAuditLogger) LogSuccess(ctx *HookExecContext) {
+	al.log(ctx, true)
+}
+
+func (al *SQLiteAuditLogger) LogFailure(ctx *HookExecContext, err error) {
+	al.log(ctx, false)
+}
+
+var (
+	sqliteMigrations = []string{
+		`create table hook_results (
+	    id integer not null primary key autoincrement,
+	    date datetime default current_timestamp,
+	    pod_id text,
+	    pod_unique_key text,
+	    hook_name string,
+	    hook_stage string,
```

Comment:
foiled by interfaces, I am. I'll have to see about wiring this up to this implementations life-cycle.